tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,419 @@
+operational cluster after graceful nodes shutdown
+"""
+
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs.ocp import OCP, wait_for_cluster_connectivity
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.longevity import start_ocp_workload
+from ocs_ci.ocs.resources import pod
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.ocs.registry import check_if_registry_stack_exists
+from ocs_ci.ocs.monitoring import check_if_monitoring_stack_exists
+from ocs_ci.framework import config
+from ocs_ci.ocs.uninstall import (
+    remove_ocp_registry_from_ocs,
+    remove_monitoring_stack_from_ocs,
+)
+from ocs_ci.utility.uninstall_openshift_logging import uninstall_cluster_logging
+from ocs_ci.framework.pytest_customization.marks import (
+    system_test,
+    polarion_id,
+    skipif_no_kms,
+    skipif_ocs_version,
+)
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs.node import (
+    get_nodes,
+)
+from ocs_ci.ocs.resources.fips import check_fips_enabled
+
+logger = logging.getLogger(__name__)
+
+
+class TestGracefulNodesShutdown(E2ETest):
+    """
+    Test uncorrupted data and operational cluster after graceful nodes shutdown
+    """
+
+    bucket_names_list = []
+
+    @pytest.fixture(autouse=True)
+    def checks(self):
+        # This test is skipped due to https://issues.redhat.com/browse/ENTMQST-3422
+
+        try:
+            check_fips_enabled()
+        except Exception as e:
+            logger.info(f"Handled prometheuous pod exception {e}")
+
+        """
+        nodes = get_nodes()
+        for node in nodes:
+            assert (
+                node.get()["status"]["allocatable"]["hugepages-2Mi"] == "64Mi"
+            ), f"Huge pages is not applied on {node.name}"
+        """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        request,
+        project_factory,
+        pv_encryption_kms_setup_factory,
+        storageclass_factory,
+        pvc_factory,
+        pod_factory,
+        pvc_clone_factory,
+        snapshot_factory,
+    ):
+        """
+        Setting up test requirements
+        """
+
+        def teardown():
+            logger.info("cleanup the environment")
+            """ cleanup logging workload """
+            sub = OCP(
+                kind=constants.SUBSCRIPTION,
+                namespace=constants.OPENSHIFT_LOGGING_NAMESPACE,
+            )
+            logging_sub = sub.get().get("items")
+            if logging_sub:
+                logger.info("Logging is configured")
+                uninstall_cluster_logging()
+
+            """ cleanup monitoring workload """
+            if check_if_monitoring_stack_exists():
+                remove_monitoring_stack_from_ocs()
+
+            """ cleanup registry workload """
+            if check_if_registry_stack_exists():
+                remove_ocp_registry_from_ocs(config.ENV_DATA["platform"])
+
+        request.addfinalizer(teardown)
+
+        logger.info("Starting the test setup")
```

Comment:
Should we add the setup related func to a setup func?

Reply:
I think if we are not achieving optimisation or resource saving thne its ok to be as it is.